docs/aws/audit/sgaudit/rules/redshift_public_instance_exists.mdx

```diff
@@ -23,6 +23,221 @@ HIPAA
+   First, identify the Redshift clusters and their current configurations.
+   ```sh
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier, PubliclyAccessible]'
+   ```
+
+2. **Modify the Cluster to Disable Public Accessibility**:
+   For each cluster that is publicly accessible, modify the cluster to disable public accessibility.
+   ```sh
+   aws redshift modify-cluster --cluster-identifier <your-cluster-identifier> --publicly-accessible false
+   ```
+
+3. **Describe Security Groups**:
+   Identify the security groups associated with your Redshift clusters to ensure they do not allow public access.
+   ```sh
+   aws ec2 describe-security-groups --query 'SecurityGroups[*].[GroupId, GroupName, IpPermissions]'
+   ```
+
+4. **Revoke Ingress Rules Allowing Public Access**:
+   For each security group that allows public access (e.g., 0.0.0.0/0), revoke the ingress rules.
+   ```sh
+   aws ec2 revoke-security-group-ingress --group-id <your-security-group-id> --protocol tcp --port 5439 --cidr 0.0.0.0/0
+   ```
+
+By following these steps, you can ensure that your Amazon Redshift clusters are not publicly accessible, thereby enhancing the security of your data.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Amazon Redshift clusters from being publicly accessible using Python scripts, you can use the AWS SDK for Python (Boto3). Below are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Describe Redshift Clusters**:
+   Use the `describe_clusters` method to get information about your Redshift clusters.
+
+4. **Modify Cluster to Disable Public Accessibility**:
+   Use the `modify_cluster` method to set the `PubliclyAccessible` attribute to `False`.
+
+Here is a Python script that demonstrates these steps:
+
+```python
+import boto3
+
+# Initialize a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+    region_name='YOUR_AWS_REGION'
+)
+
+# Create a Redshift client
+redshift_client = session.client('redshift')
+
+# Describe Redshift clusters
+clusters = redshift_client.describe_clusters()
+
+# Iterate through clusters and disable public accessibility if enabled
+for cluster in clusters['Clusters']:
+    cluster_id = cluster['ClusterIdentifier']
+    if cluster['PubliclyAccessible']:
+        print(f"Cluster {cluster_id} is publicly accessible. Disabling public access...")
+        redshift_client.modify_cluster(
+            ClusterIdentifier=cluster_id,
+            PubliclyAccessible=False
+        )
+        print(f"Public access disabled for cluster {cluster_id}.")
+    else:
+        print(f"Cluster {cluster_id} is not publicly accessible.")
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>